pkg/inject/sidecar_builder.go

```diff
@@ -190,6 +195,17 @@ func buildEnvoySidecar(vars EnvoyTemplateVariables, env map[string]string) (core
 		},
 	}
 
+	if vars.WaitUntilProxyReady {
+		envoy.Lifecycle.PostStart = &corev1.Handler{
+			Exec: &corev1.ExecAction{Command: []string{
+				"sh", "-c", fmt.Sprintf("if [[ $(/usr/bin/envoy --version) =~ ([0-9]+)\\.([0-9]+)\\.([0-9]+)-appmesh\\.([0-9]+) && "+
+					"${BASH_REMATCH[1]} -gt 2 || ${BASH_REMATCH[2]} -gt 22 || (${BASH_REMATCH[3]} -eq 2 && ${BASH_REMATCH[4]} -gt 0) ]]; "+
```

Comment:
Is it possible for there to be a 1.22.3 release? I see 1.21.4 came out after 1.22.2.

Reply:
im not sure why `1.21.4.0` is listed as after `1.22.2.0` but the most recent version before `1.22.2.1` is suppose to be `1.22.2.0` if it doesnt follow incrementing semantic versioning... then it becomes much harder to maintain imo. but regardless i think you're right we should change this to:

```
${BASH_REMATCH[3]} -gt 2 || ${BASH_REMATCH[4]} -gt 0
```

what do you think joe?

Reply:
talked offline-- joe let me know some examples where my condition would not work as intended to only support v1.22.2.1+. should be addressed now